src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -107,137 +113,263 @@
+                    entry(SecureRandom.class, "SecureRandom", "Legacy",
+                            0, 0, 0, Integer.MAX_VALUE,
+                            INSTANTIABLE | BYTE_ARRAY_SEED | STOCHASTIC),
+                    entry(Random.class, "Random", "Legacy",
+                            48, 0, 0, 0,
+                            INSTANTIABLE | LONG_SEED),
+                    entry(SplittableRandom.class, "SplittableRandom", "Legacy",
+                            64, 0, 0, 1,
+                            INSTANTIABLE | LONG_SEED),
+                    entry(L32X64MixRandom.class, "L32X64MixRandom", "LXM",
+                            64, 1, 32, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(L64X128MixRandom.class, "L64X128MixRandom", "LXM",
+                            128, 1, 64, 2,
+                            ALL_CONSTRUCTORS),
+                    entry(L64X128StarStarRandom.class, "L64X128StarStarRandom", "LXM",
+                            128, 1, 64, 2,
+                            ALL_CONSTRUCTORS),
+                    entry(L64X256MixRandom.class, "L64X256MixRandom", "LXM",
+                            256, 1, 64, 4,
+                            ALL_CONSTRUCTORS),
+                    entry(L64X1024MixRandom.class, "L64X1024MixRandom", "LXM",
+                            1024, 1, 64, 16,
+                            ALL_CONSTRUCTORS),
+                    entry(L128X128MixRandom.class, "L128X128MixRandom", "LXM",
+                            128, 1, 128, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(L128X256MixRandom.class, "L128X256MixRandom", "LXM",
+                            256, 1, 128, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(L128X1024MixRandom.class, "L128X1024MixRandom", "LXM",
+                            1024, 1, 128, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(Xoroshiro128PlusPlus.class, "Xoroshiro128PlusPlus", "Xoroshiro",
+                            128, 1, 0, 1,
+                            ALL_CONSTRUCTORS),
+                    entry(Xoshiro256PlusPlus.class, "Xoshiro256PlusPlus", "Xoshiro",
+                            256, 1, 0, 3,
+                            ALL_CONSTRUCTORS),
+                    entry(ThreadLocalRandom.class, "ThreadLocalRandom", "Legacy",
+                            64, 0, 0, 1,
+                            0)
+            );
+        }
+
+        private static Map.Entry<String, RandomGeneratorProperties> entry(
+                Class<? extends RandomGenerator> rgClass, String name, String group,
+                int i, int j, int k, int equidistribution,
+                int flags) {
+            return Map.entry(name,
+                    new RandomGeneratorProperties(rgClass, name, group,
+                            i, j, k, equidistribution,
+                            flags | (rgClass.isAnnotationPresent(Deprecated.class) ? DEPRECATED : 0)));
```

Comment:
Hello Raffaello, this is the final remaining reflection usage and even this I think isn't required now that all the random generator implementations reside within java.base as an implementation detail.

I think we should just skip this annotation check here and set `DEPRECATED` bit on the `flags` to `0` for all implementations. When/if we do deprecate any of the random generators, we can just come here and switch that bit to on for the specific random generator when instantiating this `RandomGeneratorProperties` record. I had a brief look at the code and the documentation in `package-info.java` of `java/util/random` and we don't mention that we rely on the `@Deprecated`  annotation to determine whether an algorithm is deprecated. I think that's a good thing.

Reply:
Yes, I thought about this the other day but decided for a bit more conservative approach, relying on the annotation.

But I agree that, since the meta-information now resides in `RandomGeneratorProperties`, we might "migrate" the deprecation status here as well.

Reply:
Since the legacy generators are public classes, they can be publicly deprecated, so in the last commit the `DEPRECATED` bit for them still relies on the annotation, which IMO is the authoritative "source of truth".

For the 10 other algorithms, which are accessible only via `RandomFactoryGenerator`, we can rely on the info in `RandomProperties`.

Reply:
The deprecation process of algorithms is unclear - whether it should be tied with the `@Deprecation` of a class that provides that algorithm. We will have more clarity if/when we do deprecate these algorithms. The specification in its current form doesn't tie it to the `@Deprecation` annotation, which is a good thing. So your proposed change in this PR looks fine to me, since it will still allow us to move away from the `@Deprecated` annotation check if we wanted to, in a subsequent release/change.